MobiusExtras/Test/EffectActionTests.swift

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2019 Spotify AB.
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import MobiusCore
+import MobiusExtras
+import Nimble
+import Quick
+
+class EffectActionTests: QuickSpec {
+    // swiftlint:disable function_body_length
+    override func spec() {
+        describe("EffectAction") {
+            enum Effect: Equatable {
+                case accepted
+                case rejected
+                case acceptedWith(associatedValue: Bool)
+            }
+
+            var effectAction: EffectAction<Effect>!
+            var actionCalled = false
+            let actionClosure: () -> Void = {
+                actionCalled = true
+            }
+
+            context("when initializing with accepted effect") {
+                beforeEach {
+                    effectAction = EffectAction(.accepted, action: actionClosure)
+                    actionCalled = false
+                }
+
+                context("when asking for effect acceptance on acceptable effect") {
+                    it("is accepted") {
+                        expect(effectAction.canAccept(.accepted)).to(beTrue())
+                        expect(actionCalled).to(beFalse())
```

Comment:
What is the reason for validating that the action hasn't been called? That seems to be outside of the scope of the test case. Not a big deal, of course, but just a bit of extra noise in the tests.

Reply:
My original intention was to put it there as a counterweight to `beTrue()` check of it in `when running the action`, but I agree it's not really needed -- I'll remove it.